File: 04-03/search/linear/linear.c
#include <stdio.h>

int linear_search(int array[], int size, int target) {
    for (int i = 0; i < 5; i++) {
        if (array[i] == target) {
            return i;
        }
    }
    return -1;
}

int main() {
    int array[] = {10, 20, 30, 40, 50};
    int size = sizeof(array) / sizeof(array[0]);
    int target;

    printf("Enter the number you want to search: ");
    scanf("%d", &target);

    int result = linear_search(array, size, target);

    if (result != -1) {
        printf("Element found at index %d\n", result);
    } else {
        printf("Element not found!\n");
    }

    return 0;
}

File: 04-03/search/linear/linear.py
def linear_search(array, target):
    for index, number in enumerate(array):
        if number == target:
            return index
        
    return -1

def main():
    array = [10, 20, 30, 40, 50]
    target = int(input("Enter the number you want to search: "))
    result = linear_search(array, target)
    
    if result != -1:
        print(f"Element found at index {result}")
    else:
        print("Element not found!")
        
if __name__ == "__main__":
    main()